apps/frontend/src/pages/contributors/libs/components/contributor-update-form/contributor-update-form.tsx

```diff
@@ -17,32 +22,53 @@ const ContributorUpdateForm = ({
 	contributor,
 	onSubmit,
 }: Properties): JSX.Element => {
-	const { gitEmails, name, projects } = contributor;
+	const { gitEmails, hiddenAt, name, projects } = contributor;
+
+	const isHidden = Boolean(hiddenAt);
 
 	const gitEmailsString = gitEmails.map((email) => email.email).join(", ");
 	const projectsString = projects.map((project) => project.name).join(", ");
 
-	const { control, errors, handleSubmit } = useAppForm<{
+	const { control, errors, handleSubmit, handleValueSet } = useAppForm<{
 		gitEmails: string;
+		hiddenAt: null | string;
+		isHidden: boolean;
 		name: string;
 		projects: string;
 	}>({
 		defaultValues: {
 			gitEmails: gitEmailsString,
+			hiddenAt,
+			isHidden,
 			name,
 			projects: projectsString,
 		},
 		validationSchema: contributorPatchValidationSchema,
 	});
 
+	const isHiddenValue = useFormWatch({
+		control,
+		name: "isHidden",
+	});
+
+	useEffect(() => {
+		handleValueSet("isHidden", isHidden);
+	}, [isHidden, handleValueSet]);
+
 	const handleFormSubmit = useCallback(
 		(event_: React.BaseSyntheticEvent): void => {
-			void handleSubmit((formData: { name: string }) => {
-				const payload: ContributorPatchRequestDto = { name: formData.name };
+			void handleSubmit((formData: { isHidden: boolean; name: string }) => {
+				const hiddenAtValue = isHiddenValue ? new Date().toISOString() : null;
```

Comment:
Why don't you use `isHidden` instead of `isHiddenValue` and `useFormWatch`?

Reply:
isHidden (formData.isHidden) resulted in undefined, will investigate it more

Reply:
Maybe we need to pass a default value? there is no need for setValue here, usual checkbox should be handled as inputs using useForm default flow